modules/tensor_mechanics/src/interfacekernels/CZMInterfaceKernel.C

```diff
@@ -0,0 +1,144 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CZMInterfaceKernel.h"
+
+registerMooseObject("TensorMechanicsApp", CZMInterfaceKernel);
+
+template <>
+InputParameters
+validParams<CZMInterfaceKernel>()
+{
+  InputParameters params = validParams<InterfaceKernel>();
+  params.addRequiredParam<unsigned int>("component",
+                                        "the component of the "
+                                        "displacement vector this kernel is working on:"
+                                        " component == 0, ==> X"
+                                        " component == 1, ==> Y"
+                                        " component == 2, ==> Z");
+
+  params.addRequiredCoupledVar("displacements", "the string containing dispalcement variables");
+
+  params.addClassDescription(
+      "Cohesive Zone Interface Kernel for cohesive zone model deping only on the disaplcment jump");
```

Comment:
"Interface kernel for use with cohesive zone models (CZMs) that compute traction as a function of the displacement jump"

Reply:
done